base/scopedvalues.jl

```diff
@@ -0,0 +1,136 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+module ScopedValues
+
+export ScopedValue, scoped
+
+"""
+    ScopedValue(x)
+
+Create a container that propagates values across dynamic scopes.
+Use [`scoped`](@ref) to create and enter a new dynamic scope.
+
+Values can only be set when entering a new dynamic scope,
+and the value referred to will be constant during the
+execution of a dynamic scope.
+
+Dynamic scopes are propagated across tasks.
+
+# Examples
+```jldoctest
+julia> const svar = ScopedValue(1);
+
+julia> svar[]
+1
+
+julia> scoped(svar => 2) do
+           svar[]
+       end
+2
```

Comment:
Would it make sense to show the value of `svar[]` again?

Reply:
Since we are returning `2` here I tought that would be sufficient.

Reply:
Sorry, I meant "show that it's back to 1 again in the current scope"